pkg/apis/controller/trials/v1alpha3/trial_types.go

```diff
@@ -42,6 +42,13 @@ type TrialSpec struct {
 	// Whether to retain the trial metrics collector CronJob object after completed.
 	// Deprecated
 	RetainMetricsCollector bool `json:"retainMetricsCollector,omitempty"`
+	// Describes how metrics will be collected
+	MetricsCollector MetricsCollectorSpec `json:"metricsCollector,omitempty"`
```

Comment:
Should we move https://github.com/kubeflow/katib/pull/745/files#diff-43075120b08ebfab109947ed650154cbR41 into the spec?

Reply:
In fact, moving "image" into trial cannot replace Collector, we need  Collector.Kind to decide how to create empty-dir volume to share files between worker container and metricsCollector container. So I think we can get image and inject it into worker Pod whiling create share volume in inject webhook logic.